### PR TITLE
Throw exception if a location is unreachable

### PR DESCRIPTION
**Pull request by dnesbitt61**
Return will be 400: Location N is unreachable.

### CODE REVIEW

File: src/thor/optimized_path.cc [hunk 1]
```diff
@@ -39,11 +39,31 @@ namespace valhalla {
...
+        if (!reachable) {
+          throw std::runtime_error("Location " + std::to_string(i) + " is unreachable");
+        }
+      }
+
+      // Set time costs to send to Optimizer.
       std::vector<float> time_costs;
-      for (size_t i = 0; i < td.size(); i++)
-        time_costs.emplace_back(static_cast<float>(td[i].time));
+      for (auto itr : td) {
```
Comment (dgearhart) [inline src/thor/optimized_path.cc]: auto& itr

File: src/thor/optimized_path.cc [hunk 2]
```diff
@@ -39,11 +39,31 @@ namespace valhalla {
...
+      for (uint32_t i = 0; i < n; i++) {
+        bool reachable = false;
+        for (uint32_t j = 0; j < n; j++) {
+          if (td[idx].time != kMaxCost && i != j) {
+            reachable = true;
+          }
+          idx++;
+        }
+        if (!reachable) {
+          throw std::runtime_error("Location " + std::to_string(i) + " is unreachable");
```
Comment (dgearhart) [inline src/thor/optimized_path.cc]: Should the description be "Location at index "?

File: src/thor/optimized_path.cc [hunk 3]
```diff
@@ -39,11 +39,31 @@ namespace valhalla {
...
+      for (uint32_t i = 0; i < n; i++) {
+        bool reachable = false;
+        for (uint32_t j = 0; j < n; j++) {
+          if (td[idx].time != kMaxCost && i != j) {
+            reachable = true;
+          }
+          idx++;
+        }
+        if (!reachable) {
+          throw std::runtime_error("Location at index " + std::to_string(i) + " is unreachable");
```
Comment (kevinkreiser) [inline src/thor/optimized_path.cc]: We'll need to add this to the docs errors page

**Comment by kevinkreiser** [comment]
:rocket:

**Comment by dgearhart** [comment]
:shipit:

**Pull request merged**